### PR TITLE
Addition of endpoints by country

### PR DESCRIPTION
**Pull request by oscardiazencora**
-Addition of Base controller
-Addition of constants
-Fixed date bug response
-Separation of logic in controllers
-Added Extension method to separate logic in starup cs

### CODE REVIEW

File: .vs/covid19-api/config/applicationhost.config [hunk 1]
```diff
@@ -0,0 +1,988 @@
+﻿<?xml version="1.0" encoding="UTF-8"?>
```
Comment (yeseniamolinab) [inline .vs/covid19-api/config/applicationhost.config:1]: Remove this file, we don't need it.
Also add it to gitignore

File: Common/ExtensionMethods/ServiceDIExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿using covid19_api.Services;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace covid19_api.Common.ExtensionMethods
+{
+    public static class ServiceDIExtensions
+    {
+        public static void AddServicesDI(this IServiceCollection serviceProvider)
```
Comment (yeseniamolinab) [inline Common/ExtensionMethods/ServiceDIExtensions.cs:8]: good call 👍

File: Controllers/ControllerFacade.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using Microsoft.AspNetCore.Mvc;
+
+namespace covid19_api.Controllers
+{
+    /// <summary>
+    /// implementation of generic facade controller
+    /// </summary>
+    /// <seealso cref="Microsoft.AspNetCore.Mvc.ControllerBase" />
+    [ApiController]
+    public class ControllerFacade : ControllerBase
```
Comment (yeseniamolinab) [inline Controllers/ControllerFacade.cs:10]: Can you explain why do you think we need this Facade?
What are the benefits?

File: Controllers/PredictionController.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        }
+
+        /// <summary>
+        /// Gets the specified country.
+        /// </summary>
+        /// <param name="country">The country.</param>
+        /// <returns></returns>
+        [HttpGet]
+        [Route(ApiConstants.CountryParam)]
+        public Task<IEnumerable<CovidPrediction>> Get([FromRoute] string country)
```
Comment (yeseniamolinab) [inline Controllers/PredictionController.cs:27]: let's keep all the endpoints asynchronous

File: Properties/launchSettings.json [hunk 5]
```diff
@@ -12,15 +12,15 @@
...
+      "launchUrl": "covid",
       "environmentVariables": {
         "ASPNETCORE_ENVIRONMENT": "Development"
       }
     },
     "covid19_api": {
       "commandName": "Project",
       "launchBrowser": true,
-      "launchUrl": "weatherforecast",
+      "launchUrl": "covid",
```
Comment (yeseniamolinab) [inline Properties/launchSettings.json:23]: 👍

File: Startup.cs [hunk 6]
```diff
@@ -22,6 +15,9 @@ public Startup(IConfiguration configuration)
             Configuration = configuration;
         }
 
+        /// <summary>
```
Comment (yeseniamolinab) [inline Startup.cs:18]: Remove this

File: covid19-api.csproj [hunk 7]
```diff
@@ -6,9 +6,11 @@
     </PropertyGroup>
 
     <ItemGroup>
+      <PackageReference Include="FluentValidation" Version="9.2.2" />
       <PackageReference Include="Flurl" Version="3.0.0" />
       <PackageReference Include="Flurl.Http" Version="3.0.0" />
       <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
+      <PackageReference Include="Pitcher" Version="2.1.1" />
```
Comment (yeseniamolinab) [inline covid19-api.csproj:13]: Remove unused package

File: covid19-api.csproj.user [hunk 8]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (yeseniamolinab) [inline covid19-api.csproj.user:1]: don't checkin this file and add it to gitignore

**Review by yeseniamolinab — CHANGES_REQUESTED** [review]
Please split this PR into 3, one PR per issue to keep track of what goes to master by issue.

Do not checkin binary files and add them to gitignore

I left several comments please take a look a them.

File: Controllers/ControllerFacade.cs (hunk 3 above)
Comment (oscardiazencora) [reply Controllers/ControllerFacade.cs:10]: sure, its a faćade patttern design , which in few works in meant to be inherith in the controllers layer to stop writting repetive code, for example the [apicontroller] attribute , others more important like if we were used to use CQRS or mediator , you can inject this type of props for every controller.

File: Controllers/PredictionController.cs (hunk 4 above)
Comment (oscardiazencora) [reply Controllers/PredictionController.cs:27]: okey... this method is returning a task , the endpoint callers will make it async by calling the method , like in a Promise in js, or did not I catch the async comment? or maybe was another function?

File: covid19-api.csproj (hunk 7 above)
Comment (oscardiazencora) [reply covid19-api.csproj:13]: yes, sure

File: Startup.cs (hunk 6 above)
Comment (oscardiazencora) [reply Startup.cs:18]: sure I forgot this

File: Controllers/PredictionController.cs (hunk 4 above)
Comment (yeseniamolinab) [reply Controllers/PredictionController.cs:27]: nvm this is cool this way

**Pull request closed without merging**